book.md

```diff
@@ -495,17 +492,160 @@ probably include, or be followed by, a section on the various sanitizers
 available (ASan, UBSan, etc).}
 
 ## Code reuse attacks
-\missingcontent{Discuss ROP, JOP, COOP and mitigations (ASLR, CFI etc)}
+
+In the early days of memory vulnerability exploitation, attackers could simply
+place shellcode\index{shellcode} of their choice in executable memory and jump
+to it. As executable-space protection became mainstream, attackers started to
+reuse code already present in an application's binary and linked libraries
+instead. A variety of different techniques to this effect came to light.
+
+The simplest of these techniques is return-to-libc [@Solar1997]. Instead of
+returning to shellcode that the attacker has injected, the return address is
+modified to return into a library function, such as `system` or `exec`. This
+technique is simpler to use when arguments are also passed on the stack and
+can therefore be controlled with the same stack buffer overflow that is used
+to modify the address.
+
+Return-to-libc attacks restrict an attacker to whole library functions. While
+this can lead to powerful attacks, it has also been demonstrated that it is
+possible to achieve arbitrary computation by combining a number of short
+instruction sequences known as **gadgets**\index{gadget}. In return-oriented
```

Comment:
I'm wondering if gadgets would be easier to understand as a concept if here it said already that gadgets are typically ending in an indirect control flow instruction, which makes it easy to have them executed one after another based on memory overwritten/controlled by the attacker?